src/embed.ts

```
@@ -86,44 +96,65 @@ function nukeLastLine(doc: Doc): Doc {
-    );
+    const encodedContent = getAttributeTextValue(snippedTagContentAttribute, node);
     let content = '';
-    if (
-        contentAttribute &&
-        Array.isArray(contentAttribute.value) &&
-        contentAttribute.value.length > 0
-    ) {
-        const encodedContent = (contentAttribute.value[0] as TextNode).data;
+
+    if (encodedContent) {
         content = Buffer.from(encodedContent, 'base64').toString('utf-8');
     }
-    node.attributes = node.attributes.filter(n => n !== contentAttribute);
-
-    const docContent = concat([hardline, nukeLastLine(textToDoc(content, { parser }))]);
-
-    return group(
-        concat([
-            '<',
-            tag,
-            indent(group(concat(path.map(childPath => childPath.call(print), 'attributes')))),
-            '>',
-            indentContent ? indent(docContent) : docContent,
-            hardline,
-            '</',
-            tag,
-            '>',
-            inline ? '' : hardline,
-        ]),
+
+    if (isNodeSupportedLanguage(node)) {
+        const indentContent = options.svelteIndentScriptAndStyle;
+
+        try {
+            const indentIfDesired = (doc: Doc) => (indentContent ? indent(doc) : doc);
+
+            return concat([
+                indentIfDesired(concat([hardline, nukeLastLine(textToDoc(content, { parser }))])),
+                hardline,
+            ]);
+        } catch (error) {
+            if (process.env.PRETTIER_DEBUG) {
+                throw error;
+            }
+
+            // We will wind up here if there is a syntax error in the embedded code. If we throw an error,
+            // prettier will try to print the node with the printer. That will fail with a hard-to-interpret
+            // error message (e.g. "Unsupported node type", referring to `<script>`).
+            // Therefore, fall back on just returning the unformatted text.
+            if (options.svelteReportSyntaxErrors) {
+                console.error(error);
```

Background on this: previously, these error were just swallowed and masked by the "unsupported node type" error. IMO, they should be reported; it's not necessarily a syntax error, it could be a bug in the plugin code and should therefore not be hidden. That's why I added a print to stderr.

However, I added a test that formats a file containing JS with a syntax error and I don't want that test to print an error on the console so I added this option to be able to hide it. It's a bit ugly that the option is visible to the user, but I see no other good way of removing the printout in during tests.

One alternative would be to introduce a logging framework. That would also be good to be able to add some debug output during printing, but it's a major change.

---

I'm against this new option. I personally don't mind if there is an error logged during testing. You could also add a hack to the tests like 

```ts
if (testfilename.includes('no-error-logs') {
   console.error = () => {};
}
```

---

Ok, I agree it's not a great solution. I do think it's important that there aren't error stacks when the tests are successful, though. It's confusing and can hide other errors.

Anyway, I replaced it with a solution that overwrites `console.error` to not log syntax errors when they are expected. That feels fairly clean.